src/api/exchange/ExchangeApiImpl.ts

```diff
@@ -0,0 +1,221 @@
+import assert from 'assert'
+
+import { DepositApiImpl } from './DepositApiImpl'
+import { ExchangeApi, PlaceOrderParams, Receipt, TxOptionalParams, StablecoinConverter, AuctionElement } from 'types'
+import StablecoinConvertedAbi from './StablecoinConverterAbi'
+import { log } from 'utils'
+
+import BN from 'bn.js'
+
+import Web3 from 'web3'
+
+const ADDRESS_WIDTH = 20 * 2
+const UINT256_WIDTH = 32 * 2
+const UINT16_WIDTH = 2 * 2
+const UINT32_WIDTH = 4 * 2
+const UINT128_WIDTH = 16 * 2
+
+const hexPattern = '[0-9a-fA-F]'
+
+const hn = (n: number): string => hexPattern + `{${n}}`
+
+// /(address)(sellTokenBalance)(buyTokenId)(sellTokenId)(validFrom)(validUntil)(priceNumerator)(priceDenominator)(remainingAmount)/g
+const orderPattern = `(?<user>${hn(ADDRESS_WIDTH)})(?<sellTokenBalance>${hn(UINT256_WIDTH)})(?<buyTokenId>${hn(
+  UINT16_WIDTH,
+)})(?<sellTokenId>${hn(UINT16_WIDTH)})(?<validFrom>${hn(UINT32_WIDTH)})(?<validUntil>${hn(
+  UINT32_WIDTH,
+)})(?<priceNumerator>${hn(UINT128_WIDTH)})(?<priceDenominator>${hn(UINT128_WIDTH)})(?<remainingAmount>${hn(
+  UINT128_WIDTH,
+)})`
+
+// decodes Orders for a userAddress if passed one
+// otherwise decodes all Orders for all users
+const decodeAuctionElements = (bytes: string, userAddress?: string): AuctionElement[] => {
```

Comment:
Can we move this to the utils?

eventually we are also moving it back to a common lib. By the way, they already had something for this in dex-contracts

Reply:
Eventually I hope we get `getOrdersForUser` on the contract and this will go away completely.
Also this function is only used in this file.
So let it stay, and then we get rid of it entirely.

Reply:
Okay, although I believe it's possible that we include it in the common js libs we want to do